src/module/Entity/src/Entity/Manager/EntityManager.php

```diff
@@ -79,32 +82,30 @@ public function getEntity($id)
     {
         $className = $this->getClassResolver()->resolveClassName('Entity\Entity\EntityInterface');
         $entity = $this->getObjectManager()->find($className, $id);
-        
+
         if (! is_object($entity)) {
             throw new Exception\EntityNotFoundException(sprintf('Entity "%d" not found.', $id));
         }
         $this->assertGranted('entity.get', $entity);
-        
+
         return $entity;
     }
 
     public function findAllUnrevised()
     {
         $entityClassName = $this->getClassResolver()->resolveClassName('Entity\Entity\EntityInterface');
-        $uuidClassName = $this->getClassResolver()->resolveClassName('Uuid\Entity\UuidInterface');
-        $revisionClassName = $this->getClassResolver()->resolveClassName('Entity\Entity\RevisionInterface');
-        
-        $results = $this->objectManager->createQueryBuilder()
-            ->select('e')
-            ->from($entityClassName, 'e')
-            ->Join($revisionClassName, 'r', 'WITH', 'e = r.repository')
-            ->Join($uuidClassName, 'u', 'WITH', 'u = r')
-            ->andWhere('r > e.currentRevision OR e.currentRevision is null')
-            ->andWhere('u.trashed = :trashed')
-            ->setParameter('trashed', false)
-            ->getQuery()
-            ->getResult();
-        
+        //TODO: unhack
+        $sql = 'SELECT DISTINCT e1_.id AS id FROM entity_revision e2_ INNER JOIN `uuid` u3_ ON e2_.id = u3_.id INNER JOIN entity e1_ ON (e1_.id = e2_.repository_id) WHERE ( e1_.current_revision_id IS NULL OR e2_.id > e1_.current_revision_id ) AND u3_.trashed = 0';
+        $q = $this->objectManager->getConnection()->prepare($sql);
+        $q->execute();
+        $unrevisedEntityIdsNested = $q->fetchAll();
+        $unrevisedEntityIds = [];
+        foreach ($unrevisedEntityIdsNested as $unrevisedEntityIdArray) {
+            $unrevisedEntityIds[] = $unrevisedEntityIdArray["id"];
+        }
+        $results = $this->getObjectManager()->getRepository($entityClassName)->findBy([
+            'id' => $unrevisedEntityIds
+        ]);
```

Comment:
This fetches the objects belonging to the id's.... because fetching the entities directly was terribly slow (as already explained)